### PR TITLE
6229 subdomainless urls dont work with ips

### LINKED ISSUE #6229: Subdomainless urls don't work with IPs
Enabling [subdomainless URLs](http://cartodb.readthedocs.org/en/latest/configuration.html?highlight=subdomainless#domainless-urls) works if you login at `localhost.lan:3000/login`, redirecting you to `http://localhost.lan:3000/user/USERNAME/dashboard/`, but if you login at the IP (`192.168.33.10:3000/login`), you get redirected to `http://192.168.33.10.localhost.lan:3000/user/USERNAME/dashboard/`.

cc @javisantana

### PR DESCRIPTION
**Pull request by juanignaciosl**
This fixes #6229.

### CODE REVIEW

**Comment by juanignaciosl** [comment]
@jgoizueta , @javisantana could you CR this, please?

**Comment by juanignaciosl** [comment]
@jgoizueta , @javisantana ping

**Comment by javisantana** [comment]
I can't say too much about the code but the solution looks good enough, not so many changes and fixes the issue.

One question: if I setup subdomainless urls and whatever.com as domain, can I access using bot ways, whaever.com and whaever.com ip address, right?

**Comment by juanignaciosl** [comment]
That kind of questions are the reason I've added you here ;-)

Yes, you can. In my tests I've kept _localhost.lan_ in the configuration (avoiding IPs there) and tried with both _localhost.lan_ and IP. `config/initializers/session_store.rb` change is caused by that, since using a fixed domain makes the cookie don't work for IP.

File: .gitignore [hunk 1]
```diff
@@ -61,3 +61,4 @@ public/test_support.csv
 public/test_guess_country.csv
 doc/manual/build
 .gem/
+.byebug_history
```
Comment (xavijam) [inline .gitignore]: What is it for?

Comment (xavijam) [inline .gitignore]: Oh, ruby 2 debugging? :dancer:

**Comment by javisantana** [comment]
:es:

File: .gitignore (hunk 1 above)
Comment (juanignaciosl) [inline .gitignore]: yup :-)

File: config/initializers/session_store.rb [hunk 2]
```diff
@@ -1,3 +1,2 @@
-CartoDB::Application.config.session_store :cookie_store, key: '_cartodb_session', secure_random: true,
-                                          domain: Cartodb.config[:session_domain], expire_after: 7.days,
-                                          httponly: true, secure: !(Rails.env.development? || Rails.env.test?)
+domain = CartoDB.subdomainless_urls? ? nil : Cartodb.config[:session_domain]
```
Comment (jgoizueta) [inline config/initializers/session_store.rb]: I guess initialization of `Cartodb.config` before being used here is guaranteed by the file naming, `01_app_config.rb` preceding `session_store.rb'. Is that explicitly stated in Rails?

**Comment by jgoizueta** [comment]
:thumbsup: I see no problem here

File: config/initializers/session_store.rb (hunk 2 above)
Comment (juanignaciosl) [inline config/initializers/session_store.rb]: Good question. Yes, [Rails guarantees it](http://guides.rubyonrails.org/v3.2.20/configuring.html#using-initializer-files): "If you have any ordering dependency in your initializers, you can control the load order by naming. For example, 01_critical.rb will be loaded before 02_normal.rb." . And numbers < alphabet: `> ["1", "a", "2"].sort  => ["1", "2", "a"]`

**Pull request merged**